common/persistence/pinot/pinot_visibility_store_test.go

```diff
@@ -0,0 +1,1588 @@
+
+	testContextTimeout = 5 * time.Second
+
+	validSearchAttr = definition.GetDefaultIndexedKeys()
+)
+
+func TestRecordWorkflowExecutionStarted(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	mockPinotClient := pnt.NewMockGenericClient(ctrl)
+	mockProducer := &mocks.KafkaProducer{}
+	mgr := NewPinotVisibilityStore(mockPinotClient, &service.Config{
+		ValidSearchAttributes:  dynamicconfig.GetMapPropertyFn(definition.GetDefaultIndexedKeys()),
+		ESIndexMaxResultWindow: dynamicconfig.GetIntPropertyFn(3),
+	}, mockProducer, testlogger.New(t))
+	visibilityStore := mgr.(*pinotVisibilityStore)
+
+	// test non-empty request fields match
+	request := &p.InternalRecordWorkflowExecutionStartedRequest{}
+	request.WorkflowID = "wid"
+	memoBytes := []byte(`test bytes`)
+	request.Memo = p.NewDataBlob(memoBytes, common.EncodingTypeThriftRW)
+
+	tests := map[string]struct {
+		request       *p.InternalRecordWorkflowExecutionStartedRequest
+		expectedError error
+	}{
+		"Case1: normal case": {
+			request:       request,
+			expectedError: nil,
+		},
+		"Case2: error case": {
+			request:       request,
+			expectedError: fmt.Errorf("error"),
+		},
+	}
+
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			assert.NotPanics(t, func() {
+				mockPinotClient.EXPECT().GetTableName().Return(testTableName).AnyTimes()
+				mockProducer.On("Publish", mock.Anything, mock.MatchedBy(func(input *indexer.PinotMessage) bool {
+					assert.Equal(t, request.WorkflowID, input.GetWorkflowID())
+					return true
+				})).Return(nil).Once()
+				mockProducer.On("Publish", mock.Anything, mock.MatchedBy(func(input *indexer.PinotMessage) bool {
+					assert.Equal(t, request.WorkflowID, input.GetWorkflowID())
+					return true
+				})).Return(fmt.Errorf("error")).Once()
+
+				err := visibilityStore.RecordWorkflowExecutionStarted(context.Background(), test.request)
+				assert.Equal(t, test.expectedError, err)
+			})
+		})
+	}
+}
+
+func TestRecordWorkflowExecutionClosed(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	mockPinotClient := pnt.NewMockGenericClient(ctrl)
+	mockProducer := &mocks.KafkaProducer{}
+	mgr := NewPinotVisibilityStore(mockPinotClient, &service.Config{
+		ValidSearchAttributes:  dynamicconfig.GetMapPropertyFn(definition.GetDefaultIndexedKeys()),
+		ESIndexMaxResultWindow: dynamicconfig.GetIntPropertyFn(3),
+	}, mockProducer, testlogger.New(t))
+	visibilityStore := mgr.(*pinotVisibilityStore)
+
+	// test non-empty request fields match
+	request := &p.InternalRecordWorkflowExecutionClosedRequest{}
+	request.WorkflowID = "wid"
+	memoBytes := []byte(`test bytes`)
+	request.Memo = p.NewDataBlob(memoBytes, common.EncodingTypeThriftRW)
+
+	tests := map[string]struct {
+		request       *p.InternalRecordWorkflowExecutionClosedRequest
+		expectedError error
+	}{
+		"Case1: normal case": {
+			request:       request,
+			expectedError: nil,
+		},
+		"Case2: error case": {
+			request:       request,
+			expectedError: fmt.Errorf("error"),
+		},
```

Comment:
these two test cases has similar request. why would 2nd one expect an error? 
looking at the test case in the table it should be clear what it's trying to achieve without depending on any other state

Reply:
Removed case 2 as it is meaningless and doesn't contribute to coverage at all.